### ISSUE TITLE
Enhancement request to allow auto-hiding of columns where all entries are None or Empty equivalent

### ISSUE THREAD

**Issue by lcorrigan** [man-group/dtale#647]
My most common use case is for viewing data in dtale is displaying decoded Pandas data from database queries.  The DB has a raw data column like a dictionary of fields:values that I unpack into individual columns.  As columns are concatenated to the data frame, any field within a row that doesn't have applicable data is set to None type.  So, after decoding, there are many None/NaN entries mixed throughout the data frame.  Regardless of unpacked fields, each row will have a common 3 fields whos' combined combination describes the possible fields that apply to those rows.  For example all rows with module: 3, parameter: 1, component: 1, would have non-None values for 3 applicable fields.  When I've filtered view to a specific combination of these, it would be ideal if the columns that contain all None/Missing/Empty data could be automatically hidden and restored when removing the filter.
When just using Jupyterlab or VSCODE, I typically do this with df.dropna(axis=1, how='all') to display a filtered copy.  Since most of my dtale work is just viewing the different values of applicable fields for filtered rows, it would make it very useful to have the (some times like 50 columns) hidden so I can see only the columns that are applicable.

**Comment by aschonfeld** [comment]
@lcorrigan so I've built something locally that will auto-hide any columns where all the values are `nan` (which would be the same as running `df.dropna(axis=1, how='all')`).  There are a couple ways you would be able to achieve this, the first would be by specifying `auto_hide_empty_columns` when calling `dtale.show`:

```
import numpy as np
import pandas as pd
import dtale

dtale.show(pd.DataFrame(dict(foo=['yes'], bar=[np.nan])), auto_hide_empty_columns=True)
```

The second would be using global configuration. Following this [documentation](https://github.com/man-group/dtale/blob/master/docs/CONFIGURATION.md) you can add the following to the [show] section for it to be applied to every dataframe you load into D-Tale without specifying parameters in your code:

```
auto_hide_empty_columns = True
```

Let me know if this sounds like it will work for you or if you have any changes that you want me to add? Thanks

**Comment by lcorrigan** [comment]
Hi @aschonfeld, I'm sorry to say I've not been using dtale that much in the past year.  I liked using dtale but found every time I used it I needed to create a bug ticket or enhancement request to get to my end goal, so I've kind of shelved it for now. If you want to implement it because you think it would enhance dtale, that is up to you but I should have probably retracted this request a few months ago (my apologies).
